### ISSUE TITLE
Provide user input for processing options

### ISSUE THREAD

**Issue by JonathanLorimer** [mikaelmello/inquire#97]
It would be great if the `options` field for `Select` and `MultiSelect` could accept a lambda as well as a vec, could probably do this via a trait, or you could just always accept a lambda and users who don't want to process the options list could just use do this `|_| opts`.

The purpose for this is that it would allow me to sort my options based on a fuzzy string match score. Alternatively you could provide a `sort` field on the struct.

**Comment by mikaelmello** [comment]
Hey Jonathan,

If I got that right, you want to update the options list based on the user's text input? Putting the options closer to the input first?

**Comment by mikaelmello** [comment]
I like the idea, still thinking about a solution that doesn't get in the way of the "as simple as possible" use-case but that still allows you to do what you want.

**Comment by JonathanLorimer** [comment]
Yeah, I am not too familiar with the internals of inquire, so this might break the existing model. The nice thing about this is that you can use the same mechanism to filter (or do any transformation based on input really!).
